canoniCAI/nodes.jac

```diff
@@ -178,22 +169,56 @@ node state {
         }
     }
 
-    can classify_intent with get_cand_intents activity {
+    can classify_intent {
         if (visitor.hoping) {
             if (visitor.overwrite_intent != "") {
                 visitor.predicted_intent = visitor.overwrite_intent;
                 visitor.intent_confidence = 1;
             } else {
-                question_emb = use.encode(visitor.question)[0];
-                for i in here.cand_intents {
-                    intent_emb = use.encode(i)[0];
-                    cos_score = vector.cosine_sim(question_emb, intent_emb);
-                    if (cos_score > visitor.intent_confidence) {
-                        visitor.intent_confidence = cos_score;
-                        visitor.predicted_intent = i;
+                if (visitor.clf_to_use == "biencoder") {
+                    ::collect_intents;
+                    # Use the model to perform inference
+                    # returns the list of context with the suitable candidates
+                    resp_data = bi_enc.infer(
+                        contexts=[visitor.query],
+                        candidates=here.cand_intents,
+                        context_type="text",
+                        candidate_type="text"
+                    );
+
+                    # Iterate through the candidate labels and their predicted scores
+                    max_score = 0;
+                    max_intent = "";
+                    pred=resp_data[0];
+
+                    for j=0 to j<pred["candidate"].length by j+=1 {
+                        if (pred["score"][j] > max_score){
+                            max_intent = pred["candidate"][j];
+                            max_score = pred["score"][j];
+                        }
                     }
+
+                    visitor.predicted_intent = max_intent;
+                    visitor.intent_confidence = max_score;
+
+                } else {
```

Comment:
A small improvement -- let's have this to be if and else if and always require an explicit type of CLF. But, at the walker input, let's give clf_to_use a default value.